architecture_docs/11_risks_and_technical_debt.md

```diff
@@ -0,0 +1,16 @@
+# Risks and Technical Debt
+
+## Configuration Processing
+
+We use a `values.yaml` file as a main input source for the Helm chart for the configuration of a Loculus instance.
+
+We leveraged the powerful templating capabilities of Helm to generate the configuration files for the individual artifacts.
+This works well, because we can distribute the mostly redundant configuration values efficiently.
+
+However, this became quite complex and hard to maintain over time.
+It is untested and hard to debug, if something goes wrong.
+It is also (as of now) mostly undocumented.
+
+Some parts of the configuration are redundant and could be simplified.
+Also, the Helm chart contains a lot of default values 
+that are not suitable for general Loculus instances and will result in unexpected behavior if not overwritten.
```

Comment:
Ena deposition was written as an optional component, however it still needs to submit all data and keep submission state. Therefore we duplicate all records and keep them in the backend db schema and the ena deposition schema - this creates unnecessary database bloat.

Potentially the ingest and ena-submission pod should be merged together as they both interact with INSDC and are optional. Additionally, ingest queries the ena deposition schema directly at the moment to ensure it does not reingest sequences that we submitted. Ideally ingest would not directly interact with the database but only with the backend - but this is just a lot of additional code and maintenance with no actual design improvement.

Reply:
I added something :+1: